src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -356,26 +356,37 @@
   environment-module-response! :- ringutils/RingResponse
   "Process the environment module information, returning a ring response to be
   propagated back up to the caller of the environment_modules endpoint."
-  [info-from-jruby :- List
-   environment :- schema/Str]
-  (let [info-as-json (module-info-from-jruby->module-info-for-json
-                       info-from-jruby environment)]
-    (middleware-utils/json-response 200 info-as-json)))
+  ([info-from-jruby :- List]
+   (let [return-data (map #(module-info-from-jruby->module-info-for-json
+                             (first (vals %))
+                             (name (first (keys %))))
+                          (sort-nested-info-maps info-from-jruby))]
```

Comment:
This is also a tricky part of this change. This gives me want I want, but surely there's a better way to handle this data and use these formatting functions in a cleaner way...? Maybe not.